nox/_option_set.py

```diff
@@ -233,8 +233,22 @@ def _finalize_args(self, args):
 
     def parse_args(self):
         parser = self.parser()
-        args = parser.parse_args()
-
+        all_args = sys.argv[1:]
+        if "--" in all_args:
+            split_index = all_args.index("--")
+            nox_args = all_args[0:split_index]
+            posargs = all_args[split_index + 1 :]
+        else:
+            nox_args = all_args
+            posargs = []
+
+        args = parser.parse_args(nox_args)
+        if args.posargs:
+            # posargs should not be present since we explicitly only parsed
+            # values before the '--'
+            parser.error(f"Unrecognized argument(s): '{' '.join(args.posargs)}'")
```

Comment:
Would it actually be possible to move this logic into the posargs finalizer?

Reply:
Yeah now that you mention it, it does look like it makes more sense there.